Add support for v3.16

NickCondron opened the pull request:
https://github.com/project-slippi/slippi-wiki/blob/master/SPEC.md#post-frame-update

Latest fields added by yours truly

hohav commented on gen/resources/frames.json:
```diff
@@ -75,7 +75,11 @@
 		{ "name": "animation_index", "type": "u32",
-			"version": "3.11" }
+			"version": "3.11" },
+		{ "name": "instance_hit_by", "type": "u16",
```
I'm thinking we should name this similarly to the existing `last_hit_by` field. Maybe `last_hit_by_instance_id`?

NickCondron replied:
`last_hit_by` isn't even true to its name because it resets to 6 (signifying no other player) after a certain number of frames. The uselessness of that field was the main reason I even went about adding those fields in the first place.

Anyway, I generally prefer the shorter name that isn't super semantic. In fact even that name could be misleading given that it resets to 0 on death. I just went with the name in the spec, but if you want to change it you can.

hohav reviewed (commented):
Thanks for this. Do you happen to have a v3.16 replay for tests?

NickCondron commented:
> Thanks for this. Do you happen to have a v3.16 replay for tests?

Added a replay and tests. Used the v3.16 replay for the simple v3.14/v3.15 tests. More involved tests for v3.16 also added.

The v3.16 test sort of tests vanilla melee behavior, but I think it is still a useful test to verify the values make sense.

hohav commented:
Thanks for this, @NickCondron. I ended up renaming it to `last_hit_by_instance`.

Did you have any trouble with the code generation?

NickCondron commented:
> Thanks for this, @NickCondron. I ended up renaming it to `last_hit_by_instance`.
> 
> Did you have any trouble with the code generation?

No. It went well. I just had to install a few dependencies.
